src/backend/distributed/worker/task_tracker.c

```diff
@@ -57,1052 +57,3 @@ int TaskTrackerDelay = 200;       /* process sleep interval in millisecs */
-{
-	sigjmp_buf local_sigjmp_buf;
-	static bool processStartUp = true;
-
-	/* Properly accept or ignore signals the postmaster might send us */
-	pqsignal(SIGHUP, TrackerSigHupHandler); /* set flag to read config file */
-	pqsignal(SIGTERM, TrackerShutdownHandler); /* request shutdown */
-
-	/* We're now ready to receive signals */
-	BackgroundWorkerUnblockSignals();
-
-	/*
-	 * Create a memory context that we will do all our work in.  We do this so
-	 * that we can reset the context during error recovery and thereby avoid
-	 * possible memory leaks.
-	 */
-	MemoryContext TaskTrackerContext = AllocSetContextCreateExtended(TopMemoryContext,
-																	 "Task Tracker",
-																	 ALLOCSET_DEFAULT_MINSIZE,
-																	 ALLOCSET_DEFAULT_INITSIZE,
-																	 ALLOCSET_DEFAULT_MAXSIZE);
-	MemoryContextSwitchTo(TaskTrackerContext);
-
-	/*
-	 * If an exception is encountered, processing resumes here. The motivation
-	 * for this code block is outlined in postgres.c, and the code itself is
-	 * heavily based on bgwriter.c.
-	 *
-	 * In most error scenarios, we will not drop here: the task tracker process
-	 * offloads all work to backend processes, and checks the completion of work
-	 * through the client executor library. We will therefore only come here if
-	 * we have inconsistencies in the shared hash and need to signal an error.
-	 */
-	if (sigsetjmp(local_sigjmp_buf, 1) != 0)
-	{
-		/* Since we are not using PG_TRY, we must reset error stack by hand */
-		error_context_stack = NULL;
-
-		/* Prevents interrupts while cleaning up */
-		HOLD_INTERRUPTS();
-
-		/* Report the error to the server log */
-		EmitErrorReport();
-
-		/*
-		 * These operations are just a minimal subset of AbortTransaction().
-		 * We do not have many resources to worry about; we only have a shared
-		 * hash and an LWLock guarding that hash.
-		 */
-		LWLockReleaseAll();
-		AtEOXact_Files(false);
-		AtEOXact_HashTables(false);
-
-		/*
-		 * Now return to normal top-level context, and clear ErrorContext for
-		 * next time.
-		 */
-		MemoryContextSwitchTo(TaskTrackerContext);
-		FlushErrorState();
-
-		/* Flush any leaked data in the top-level context */
-		MemoryContextResetAndDeleteChildren(TaskTrackerContext);
-
-		/* Now we can allow interrupts again */
-		RESUME_INTERRUPTS();
-
-		/*
-		 * Sleep at least 1 second after any error.  A write error is likely to
-		 * be repeated, and we don't want to be filling the error logs as fast
-		 * as we can.
-		 */
-		pg_usleep(1000000L);
-	}
-
-	/* We can now handle ereport(ERROR) */
-	PG_exception_stack = &local_sigjmp_buf;
-
-	/*
-	 * We run validation and cache cleanup functions as this process is starting
-	 * up. If these functions throw an error, we won't try running them again.
-	 */
-	if (processStartUp)
-	{
-		processStartUp = false;
-
-		/* clean up old files in the job cache */
-		TrackerCleanupJobDirectories();
-
-		/* clean up schemas in the job cache */
-		TrackerCleanupJobSchemas();
```

Comment:
We'll need to run these function calls on start-up as long as we use the old re-partitioning functions.

Reply:
Makes sense, I guess we should rename them and use in the startup then right?

Reply:
I looked at the methods now.

`TrackerCleanupJobDirectories` can be renamed and used.
`TrackerCleanupJobSchemas` this currently cannot be used, and it seems that this function isn't doing anything currently, I guess this is also addressed with #3946. However, that method is task tracker specific, and we will need to write a similar method, a more generic one. Something that basically calls `worker_cleanup_job_schema_cache`.